Tests/SwiftSyntaxMacroExpansionTest/AttributeRemoverTests.swift

```diff
@@ -10,505 +10,484 @@
 //
 //===----------------------------------------------------------------------===//
 
-//==========================================================================//
-// IMPORTANT: The macros defined in this file are intended to test the      //
-// behavior of MacroSystem. Many of them do not serve as good examples of   //
-// how macros should be written. In particular, they often lack error       //
-// handling because it is not needed in the few test cases in which these   //
-// macros are invoked.                                                      //
-//==========================================================================//
-
-import SwiftDiagnostics
+import SwiftParser
 import SwiftSyntax
-import SwiftSyntaxMacroExpansion
-import SwiftSyntaxMacros
-import SwiftSyntaxMacrosTestSupport
 import XCTest
-
-fileprivate struct NoOpPeerMacro: PeerMacro {
-  static func expansion(
-    of node: AttributeSyntax,
-    providingPeersOf declaration: some DeclSyntaxProtocol,
-    in context: some MacroExpansionContext
-  ) throws -> [DeclSyntax] {
-    return []
+import _SwiftSyntaxTestSupport
+@_spi(AttributeRemover) import SwiftSyntaxMacroExpansion
+
+fileprivate func assertSyntaxRemovingTestAttributes(
+  of originalSource: () throws -> String,
+  reduction expectedReducedSource: () -> String,
+  file: StaticString = #filePath,
+  line: UInt = #line
+) {
+  let attributesToRemove = ["@Test"].map(AttributeSyntax.init(stringLiteral:))
+
+  do {
+    //  let reducedSource = WrappedAttributeRemover(attributesToRemove: attributesToRemove)
+    let reducedSource = AttributeRemover(
+      attributesToRemove: attributesToRemove,
+      attributeEquals: { $0.trimmedDescription == $1.trimmedDescription }
+    )
+      .rewrite(
+        Parser.parse(
+          source: try originalSource()
+        )
+      )
+
+    assertStringsEqualWithDiff(
+      reducedSource.description.trimmingCharacters(in: .newlines),
+      expectedReducedSource().trimmingCharacters(in: .newlines),
```

Comment:
Do we need the `.trimmingCharacters(in: .newlines)` here?

Reply:
Trimming is needed on `reducedSource` for the tests to pass. IMO, that means `AttributeRemover` needs to be fixed, as this at least undesirable if not a bug. (This behavior was previously masked by `assertMacroExpansion`.)

I originally included trimming newlines in order to mimic `assertMacroExpansion`&rsquo;s behavior: https://github.com/apple/swift-syntax/blob/f9744529b24e6a118ca29c12dfdc1103a3738819/Sources/SwiftSyntaxMacrosTestSupport/Assertions.swift#L295-L296

Without trimming newlines on the reduced source, 5 tests fail due to an extra leading newline when the removed attribute has no proceeding tokens: e.g.:

```swift
assertSyntaxRemovingTestAttributes {
  """
  @Test
  var x: Int
  """
} reduction: {
  "var x: Int"
}
//failed - Attribute removal did not produce the expected reduced source
//+
// var x: Int
//
//Actual reduced source:
//
//var x: Int
```

and:

```swift
assertSyntaxRemovingTestAttributes {
  """
  @Test
  /* comment */
  var value: Int
  """
} reduction: {
  """
  /* comment */
  var value: Int
  """
}
//failed - Attribute removal did not produce the expected reduced source
//+
// /* comment */
// var value: Int
//
//Actual reduced source:
//
///* comment */
//var value: Int
```

I'm leaning towards getting rid of the newline trimming in the assert helper for now, and then fixing the tests that fail because of that by including the unwanted leading newlines in the expected output. I'll include some comments to draw attention to these not-so-great test outputs. Once this PR lands, I'll open an issue so this can be fixed.

My reasoning is that, before this PR, it's hard to replicate the bad behavior, and I'm not sure that the fix will be trivial.

Thoughts?

Reply:
> I'm leaning towards getting rid of the newline trimming in the assert helper for now, and then fixing the tests that fail because of that by including the unwanted leading newlines in the expected output. I'll include some comments to draw attention to these not-so-great test outputs. Once this PR lands, I'll open an issue so this can be fixed.

That sounds good to me 👍🏽 Could you also file an issue that we should fix `AttributeRemover` to remove the empty newline in this case?